packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -77,19 +82,31 @@ export class VaDate {
   })
   dateBlur: EventEmitter;
 
-  private handleDateBlur = (event: FocusEvent) => {
-    const [year, month, day] = (this.value || '')
-      .split('-')
-      .map(val => parseInt(val));
+  /**
+   * Set the value prop as an ISO-8601 date using provided arguments.
+   * Strips trailing hyphens and sets date to be null if the
+   * date values are all NaNs.
+   */
+  private setValue(year: number, month: number, day: number): void {
```

Comment:
I moved the `this.value = ` assignment into this helper function so that I could centralize stripping the trailing `-` from the `value` prop so we don't get this:

`2000-01-`